### ISSUE TITLE
Return type not detected when there is an error in the docblock

### ISSUE THREAD

**Issue by devbanana** [vimeo/psalm#6596]
I'm using a library, [vfsStream](https://github.com/bovigo/vfsStream). But one of the library's methods has an error in its docblock, and it causes psalm to not detect the return type.

On [line 92 of vfsStreamFile](https://github.com/bovigo/vfsStream/blob/v1.6.10/src/main/php/org/bovigo/vfs/vfsStreamFile.php#L92) in that library, it has:

`     * @param   string]FileContent  $content`

And because of this, the `@return` on the next line isn't parsed, which results in a mixed assignment error in psalm.

Here's code that demonstrates it:

```php
$file = vfsStream::newFile('foo.txt')->withContent('bar');
```

And here's a generalized example. There are several errors but see the MixedAssignment error on line 13.

https://psalm.dev/r/4b2a2ea191

**Comment by orklah** [comment]
Building and maintaining a phpdoc parser that would work even when docblock has a syntax error seems to be a really big change to fix a few cases like this.

Most of the time, docblocks can be fixed by the user so it's not that often we have this case.

The best way to proceed would be to do a PR to the library. If not possible, creating stubs would fix the issue

**Comment by devbanana** [comment]
Thanks, that makes sense. I didn't know how it works under the surface so didn't know if it was feasible. I understand why it'd be difficult.
